src/util/longbits.js

```diff
@@ -36,23 +36,48 @@ function LongBits(lo, hi) {
  */
 var zero = LongBits.zero = new LongBits(0, 0);
 
-zero.toNumber = function() { return 0; };
+zero.toBigInt = function() { return 0n; };
 zero.zzEncode = zero.zzDecode = function() { return this; };
 zero.length = function() { return 1; };
 
+const TWO_32 = 4294967296n;
+
 /**
- * Zero hash.
- * @memberof util.LongBits
- * @type {string}
+ * Constructs new long bits from the specified number.
+ * @param {number} value Value
+ * @returns {util.LongBits} Instance
  */
-var zeroHash = LongBits.zeroHash = "\0\0\0\0\0\0\0\0";
+LongBits.fromBigInt = function fromNumber(value) {
+    value = BigInt(value);
+    if (value === 0n)
+        return zero;
+
+    var negative = value < 0;
+    if (negative) {
+        value = -value;
+    }
+    var hi = Number(value >> 32n) | 0;
+    var lo = Number(value - ( BigInt(hi) << 32n ) ) | 0;
```

Comment:
I've been unable to correctly encode uint64 fields with the max field value of `18446744073709551615n` without this change:

```suggestion
    var hi = Number(value >> 32n);
    var lo = Number(value - ( BigInt(hi) << 32n ) );
```

Reply:
I fixed it by https://github.com/juzibot/protobuf.js/pull/7